### ISSUE TITLE
finalfusion on Apple Silicon

### ISSUE THREAD

**Issue by danieldk** [finalfusion/finalfusion-utils#67]
finalfusion seems to work on Apple Silicon. I am using an x86_64 toolchain under Rosetta 2 because for some other projects the dependencies are not available yet for Apple Silicon (e.g. libtorch). Then I cross-compile for Apple Silicon:

```
$ rustup target add aarch64-apple-darwin
$ cargo build --target=aarch64-apple-darwin --release
$ file target/aarch64-apple-darwin/release/finalfusion
target/aarch64-apple-darwin/release/finalfusion: Mach-O 64-bit executable arm64
$ target/aarch64-apple-darwin/release/finalfusion
zsh: killed     target/aarch64-apple-darwin/release/finalfusion
```

Turns out that this is a known problem with [code signing in cross compilation](https://stackoverflow.com/questions/64830671/why-does-my-native-arm64-application-built-using-an-x86-64-build-system-fail-to).

```
# Sign with ad-hoc signing
$ codesign -s - target/aarch64-apple-darwin/release/finalfusion
$ target/aarch64-apple-darwin/release/finalfusion             
finalfusion 

USAGE:
    finalfusion <SUBCOMMAND>
```

Open questions:

* Can we easily generate a universal binary?
* Maybe we should hook this up to CI to build macOS releases. I think we could even cross-compile with the current x86_64 builders as long as Xcode is new enough.

**Comment by danieldk** [comment]
> Can we easily generate a universal binary?

This turns out to be easy:

```
$ cargo build --target=aarch64-apple-darwin --release
$ cargo build --target=x86_64-apple-darwin --release
$ lipo -create -output finalfusion target/aarch64-apple-darwin/release/finalfusion target/x86_64-apple-darwin/release/finalfusion
```

**Comment by danieldk** [comment]
See #88.

**Comment by danieldk** [comment]
Fixed by #88, we are building universal macOS binaries for releases now.
